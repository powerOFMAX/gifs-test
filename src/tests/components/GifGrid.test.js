import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing <GifGrid />', () => {
  test('should match snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={'one punch'} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show images on useFetchGifs done', () => {
    const gifs = [{ id: 'ABC', url: 'https://localhost/sth.jpg', title: 'cualquiercosa'}]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={'one punch'} />);
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })   
  
});
