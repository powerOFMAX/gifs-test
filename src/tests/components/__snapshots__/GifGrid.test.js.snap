// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGrid /> should match snapshot 1`] = `
<Fragment>
  <h3>
    one punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
